### PR TITLE
Honor tags on loadRPC apidef load code paths

### PR DESCRIPTION
**Pull request by titpetric**
## Description

There are two paths to load API, one is FromDashboard, and one is FromRPC. Gateway tags have been correctly implemented for Dashboard, but not for RPC (in use with MDCB). The commit does two things:

- renames a nestedApiDefinition structure wrapper which doesn't apply only to dashboard anymore,
- adds api filtering into FromRPC so that we honor the gateway segment tags settings

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring or add test (improvements in base code or adds test coverage to functionality)

## Checklist

- [ ] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). If pulling from your own
      fork, don't request your `master`!
- [ ] Make sure you are making a pull request against the **`master` branch** (left side). Also, you should start
      *your branch* off *our latest `master`*.
- [ ] My change requires a change to the documentation.
  - [ ] If you've changed APIs, describe what needs to be updated in the documentation.
  - [ ] If new config option added, ensure that it can be set via ENV variable
- [ ] I have updated the documentation accordingly.
- [ ] Modules and vendor dependencies have been updated; run `go mod tidy && go mod vendor`
- [ ] When updating library version must provide reason/explanation for this update.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] Check your code additions will not fail linting checks:
  - [ ] `go fmt -s`
  - [ ] `go vet`

### CODE REVIEW

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4187/merge
Commit:  
Triggered by: pull_request (@titpetric)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2676210611)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4187/merge
Commit: 8645749722d107a001f62d1d7e5223a1c3a58dc2 
Triggered by: pull_request (@titpetric)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2676359841)

**Review by jeffy-mathew — APPROVED** [review]

**Review by buger — APPROVED** [review]
Overall looks clean, and I would put approval. So it depends on urgency on this PR but this comes without test, but I do get that testing MDCB hard, but in this case it is totally possible! 
Check `TestSyncAPISpecsRPCSuccess` we do have RPC mock. 

So basically you mock `GetApiDefinitions` function. Set segment tags config, and after you run `ts.Gw.LoadDefinitionsFromRPCBackup()` and check count of APIs.

**Review by furkansenharputlu — APPROVED** [review]

**Comment by furkansenharputlu** [comment]
Lgtm, also let’s handle Leo’s comments

File: gateway/api_definition.go [hunk 1]
```diff
@@ -539,16 +539,25 @@ func (a APIDefinitionLoader) FromRPC(orgId string, gw *Gateway) ([]*APISpec, err
...
+	}
+
+	gwConfig := a.Gw.GetConfig()
+
+	// Extract tagged entries only
+	apiDefs := list.filter(gwConfig.DBAppConfOptions.NodeIsSegmented, gwConfig.DBAppConfOptions.Tags...)
+
 	var specs []*APISpec
 	for _, def := range apiDefs {
 		def.DecodeFromDB()
```
Comment (titpetric) [inline gateway/api_definition.go:558]: @furkansenharputlu this line smells a bit, the RPC is a request/response api - why do we need DecodeFromDB?

Comment (titpetric) [reply gateway/api_definition.go:558]: @furkansenharputlu I just got your comment (was editing mine for more context). The RPC definitions are fed from a dashboard RPC response, so this function does not have a database use context.

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4187/merge
Commit: 525c0803435220701efcbba36547e9150df92e76 
Triggered by: pull_request (@titpetric)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2703688433)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4187/merge
Commit: 014468e7e89a2973577b1082ad4c0e43f33c9e63 
Triggered by: pull_request (@titpetric)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2703760346)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4187/merge
Commit: d040329a31cafe49410630ceff46c77a73126a04 
Triggered by: pull_request (@titpetric)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2703779436)

**Comment by titpetric** [comment]
@buger I've just added enough unit tests to ensure my changes are hit by code coverage. If you'd like to see a FromRPC integration test/mock, that does add to the complexity a bit + integration tests take longer than unit tests. Let me know and I'll loop through JIRA to open a ticket for it, but I would argue that it's not necessary;

If we really want to test LoadRPC, then what we need is an e2e test together with a dashboard instance.

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/heads/master
Commit: 5d26bccaae29200d0cd2b052d43d172fb24a98b8 Honor tags on loadRPC apidef load code paths (#4187)

* Honor tags on loadRPC apidef load code paths

* Fix test due to struct rename

* Add tests for code coverage
Triggered by: push (@titpetric)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2704069893)

**Comment by titpetric** [comment]
I'm currently blocking this from merge to the release branches, as a separate panic has been raised with @furkansenharputlu - I've suggested Maciej verifies against the master branch first, where we hopefully won't be hitting a panic over the DecodeFromDB line.

**Pull request merged**